src/main/java/com/opendxl/databus/common/internal/adapter/DatabusProducerJSONRecordAdapter.java

```diff
@@ -0,0 +1,102 @@
+/*---------------------------------------------------------------------------*
+ * Copyright (c) 2019 McAfee, LLC - All Rights Reserved.                     *
+ *---------------------------------------------------------------------------*/
+
+package com.opendxl.databus.common.internal.adapter;
+
+import com.opendxl.databus.common.internal.builder.TopicNameBuilder;
+import com.opendxl.databus.common.internal.util.HeaderInternalField;
+import com.opendxl.databus.entities.MessagePayload;
+import com.opendxl.databus.producer.ProducerRecord;
+import com.opendxl.databus.serialization.Serializer;
+import com.opendxl.databus.common.MessageFormat;
+
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.lang.StringUtils;
+
+import org.apache.kafka.common.header.Header;
+import org.apache.kafka.common.header.internals.RecordHeader;
+
+/**
+ * Adapter for Databus producer record in JSON message format. The incoming message headers are set as kafka message \
+ * headers. Message payload is stored as is in JSON format in kafka message payload.
+ *
+ * @param <P> payload's type
+ */
+public final class DatabusProducerJSONRecordAdapter<P>
+                implements Adapter<ProducerRecord, org.apache.kafka.clients.producer.ProducerRecord<String, byte[]>> {
+
+        /**
+         * The message Serializer.
+         */
+        private final Serializer<P> messageSerializer;
+
+        /**
+         * Constructor
+         *
+         * @param messageSerializer a {@link Serializer} instance used for Serializing
+         *                          the payload.
+         */
+        public DatabusProducerJSONRecordAdapter(final Serializer<P> messageSerializer) {
+                this.messageSerializer = messageSerializer;
+        }
+
+        /**
+         * Adapter pattern implementation for DatabusProducerRecord instance.
+         * Adapts a ProducerRecord to a DatabusProducerRecord instance.
+         *
+         * @param sourceProducerRecord a {@link ProducerRecord} instance.
+         * @return a DatabusProducerRecord with a String key and a DatabusMessage value.
+         */
+        @Override
+        public org.apache.kafka.clients.producer.ProducerRecord<String, byte[]> adapt(
+                        final ProducerRecord sourceProducerRecord) {
+
+                Map<String, String> headers = sourceProducerRecord.getHeaders().getAll();
+                final List<Header> kafkaHeaders = new ArrayList<>();
+
+                for (final String key : headers.keySet()) {
+                        String val = headers.get(key);
+                        if (null == val) {
```

Comment:
If we are not getting value for a key then why we are adding empty value for that key ?

Reply:
The null check is removed as the Headers class  checks for null and inserts empty value